test/blackbox-tests/test-cases/foreign_cxx_library.t/run.t

```diff
@@ -0,0 +1,2 @@
+Building a cxx library should run ocamlmklib with "-lstdc++".
+  $ dune build
```

Comment:
How should I check that this test succeeds?

Reply:
If it fails a bunch of `undefined symbols` such as `std::__1::cout` errors are printed by the linker.
I will make that clear in the test description. Also the flag is not always `-stdc++`, it is `-lc++` for clang.

Also there is no specific link flag for c++ compilation with MSVC so this test won't fail on windows.

Reply:
Ack. I'm not sure we run the testsuite on Windows BTW.

Reply:
Just one last thing: I was looking at the commits individually. I was expecting to see the test fail at first and then observe subsequent commits fix it, but it seems that it was succeeding from the start. Is that the case?

Reply:
Yes they are indeed failing but the error message was not promoted, do you want me to edit it ?

Reply:
That's alright, I just wanted to check it was the case. For next time, please do promote it. It's good practice to first demonstrate the problem and then demonstrate that it is fixed. Plus it's good if the testsuite passes for each commit.